Bottom: 910d0f348176c17c1b92e4252c59660a2e782db3
Top:    70bd85a9055550edb3e2ceadaa404ed07b5fcb4c
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2013-01-03 19:10:06 +0000

find.c: OOPS, turns out we get an infinite loop with the last commit...

We need to use a flag variable to change the flag we're playing with,
otherwise we keep going around and around.

No doubt there is a nicer way to fix this, but I'm not sure I completely
understand the original intention of the code, or its exact requirements.


---

diff --git a/src/find.c b/src/find.c
index 1f49753..184e001 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3333,6 +3333,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
   long int *object_id_list;
   int *object_type_list;
   DrcViolationType *violation;
+  int flag;
 
   if (PCB->Shrink != 0)
     {
@@ -3395,10 +3396,11 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
   ListStart (What, ptr1, ptr2, ptr3, SELECTEDFLAG);
   DoIt (SELECTEDFLAG, true, false);
   DumpList ();
-  ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
+  flag = FOUNDFLAG;
+  ListStart (What, ptr1, ptr2, ptr3, flag);
   Bloat = PCB->Bloat;
   drc = true;
-  while (DoIt (FOUNDFLAG, true, false))
+  while (DoIt (flag, true, false))
     {
       DumpList ();
       /* make the flag changes undoable */
@@ -3439,13 +3441,14 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       IncrementUndoSerialNumber ();
       Undo (true);
       /* highlight the rest of the encroaching net so it's not reported again */
+      flag = FOUNDFLAG | SELECTEDFLAG;
       Bloat = 0;
-      ListStart (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, FOUNDFLAG | SELECTEDFLAG);
-      DoIt (FOUNDFLAG | SELECTEDFLAG, true, true);
+      ListStart (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag);
+      DoIt (flag, true, true);
       DumpList ();
       drc = true;
       Bloat = PCB->Bloat;
-      ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG | SELECTEDFLAG);
+      ListStart (What, ptr1, ptr2, ptr3, flag);
     }
   drc = false;
   DumpList ();
